src/Core/LoginConfig/DependencyInjection/LoginConfigCompilerPass.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\LoginConfig\DependencyInjection;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\LoginConfig\ConfigBuilder\LoginConfigService;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @internal
+ */
+#[Package('core')]
+class LoginConfigCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition(LoginConfigService::class)) {
+            return;
+        }
+
+        $definition = $container->getDefinition(LoginConfigService::class);
+        $taggedServices = $container->findTaggedServiceIds('shopware.login.config.handler');
+
+        foreach ($taggedServices as $id => $tags) {
+            foreach ($tags as $attributes) {
+                $definition->addMethodCall('addHandler', [
```

Comment:
Ah here `addHandler` is called. Why not injecting these tagged services in the constructor?

Reply:
The reason is, if i do it this was, i have a key which i can set and directly find the handler. No iteration to find the handler. Just a direct check

Reply:
That should also be possible without compiler pass, if I'm not mistaken. See
https://symfony.com/doc/current/service_container/tags.html#tagged-services-with-index

Reply:
Works